packages/types/src/getters/transaction.ts

```diff
@@ -0,0 +1,14 @@
+import { createGetter } from './utils/create-getter';
+import { Transaction } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1/transaction_pb';
+import { Swap } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/dex/v1/dex_pb';
+
+const getSwap = createGetter(
+  (transaction?: Transaction) =>
+    transaction?.body?.actions.find(action => action.action.case === 'swap')?.action.value as
+      | Swap
+      | undefined,
+);
+
+const getCommitment = createGetter((swap?: Swap) => swap?.body?.payload?.commitment);
```

Comment:
I meant to put this in the README for getters, but like with the translators, I was trying to organize the files by the type that you pass to the getter. So, all getters in this file should accept a `Transaction` as their argument. Since `getCommitment` accepts a `Swap`, it should be in a file called `swap.ts`. (Same goes for the various types in the `swaps.ts` file above — `getSwapRecordCommitment()` should go in a file called `swap-record.ts` since it accepts a `SwapRecord`, etc.)

Not a blocker — I'd leave this up to you. I just would personally keep it organized that way because we'll probably end up with dozens of getters before long, and we'll always know where to look for them if they're organized by the type they accept as their argument.